specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json

```diff
@@ -588,6 +635,28 @@
         }
       }
     },
+    "PatchTrackedResource": {

```

Comment:
It is unclear to me why `PatchTrackedResource` and `PatchVault` were added, they seem to be identical to their non-patch counterparts.  Can you explain why it is necessary or desirable to have both?

Reply:
Hi @marstr ,

PatchTrackedResource doesn't have "location" as required property as opposed to its counter part TrackedResource. This was the reason(All the top level properties should be optional) we had to introduce a new model for PATCH call: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.RecoveryServices/vaults/{vaultName}" . This was a swagger violation and hence we updated it. This solution was suggested by @amarzavery 


Here is the exact violation you can refer to:


ERROR   (PatchBodyParametersSchema/R2016/RPCViolation): Properties of a PATCH request   body must not be required. PATCH operation: 'Vaults_Update' Model Definition:   'TrackedResource' Property: 'location'
--
- file:///E:/src/azure-rest-api-specs/specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/2016-06-01/vaults.json:569:6   ($.definitions.TrackedResource.required)

Reply:
Thanks for this explanation, it is exactly what I was looking for.